### DISCUSSION TITLE
Netzwerkschnittstelle verwenden (Olimex ESP32-GATEWAY)

### DISCUSSION THREAD

**Discussion by Nonkonformist92** [tbnobody/OpenDTU#512] [Q&A]
Hallo, liebe Community!

Auf dieses tolle Projekt bin ich über das c't Magazin 24/2022 aufmerksam geworden. 

Ich besitze einen Hoymiles HM-600 Microwechselrichter und möchte diesen gerne mittels OpenDTU über einen ESP32-Mikrocontroller mit Ethernet-Schnittstelle auslesen. Hierzu habe ich einen  [Olimex ESP32-GATEWAY](https://www.olimex.com/Products/IoT/ESP32/ESP32-GATEWAY/open-source-hardware) erworben. 

Mein Problem:

Das Flashen von OpenDTU mittels Visual Studio Code konnte binnen weniger Minuten auf dem Olimex ESP32-GATEWAY erfolgreich abgeschlossen werden, allerdings ist lediglich die WLAN- und nicht die Ethernet-Schnittstelle erreichbar. 

In Visual Studio Code habe ich in der "platformio.ini" in Zeile 12 "default_envs =" den Wert "olimex_esp32_evb" eingetragen. Scheinbar ist dieser Eintrag für den Olimex ESP32-GATEWAY jedoch nicht korrekt. 

Welchen Wert muss ich konfigurieren, um die Ethernet-Schnittstelle des Mikrocontrollers verwenden zu können? Über Unterstützung würde ich mich freuen! Vielen Dank im Voraus!

**Comment by ThatRed** [comment]
Stell mal bitte den Orginalzustand der platformio.ini wieder her und öffne die platformio_override.ini. Füge dort unter 

; you can define your personal board and/or settings here

folgendes hinzu:

[env:olimex_esp32_gateway]
; https://www.olimex.com/Products/IoT/ESP32/ESP32-GATEWAY/open-source-hardware
board = esp32-gateway
build_flags = ${env.build_flags}
    -DHOYMILES_PIN_MISO=15
    -DHOYMILES_PIN_MOSI=2
    -DHOYMILES_PIN_SCLK=14
    -DHOYMILES_PIN_IRQ=13
    -DHOYMILES_PIN_CE=16
    -DHOYMILES_PIN_CS=17
    -DOPENDTU_ETHERNET

Gegebenenfalls müssen noch die Ports darüber angepasst werden.

Datei speichern und unter "platformio, project tasks", zum refresh den runden Pfeil anklicken. Danach sollte das Board (ganz unten?) auswählbar sein.

**Comment by muibusan** [comment]
Hallo @Nonkonformist92, ich weiß nicht, ob das an OpenDTU hängt?! 

Wie versorgst Du denn Dein Board mit Strom? Wenn Du Dein Board über den Micro-USB-Anschluß per Strom versorgst, reicht der u.U. nicht für die Ethernet-Anbinding aus?! Nur eine Vermutung.

Ich habe zwar nicht das Olimex ESP32-GATEWAY, sondern das [Olimex ESP32-EVB](https://www.olimex.com/Products/IoT/ESP32/ESP32-EVB/open-source-hardware) - wie Deines auch mit Ethernet-Anschluß. 

Nach dem erfolgreichen Flashen habe ich es per 5V/1A-Netzteil an den Strom angeschlossen und das Ethernet-Kabel eingesteckt. 

Ich konnte dann direkt übers LAN-Kabel auf das Board bzw. auf OpenDTU zugreifen. Natürlich nicht über die standardmäßige 192.168.1.4, sondern über die IP, die der Router per DHCP dem Board zugewiesen hat. Da mußt Du mal in Deinem Router nachschauen, welche IP er Deinem Board zuweist, nachdem Du es angesteckt hast. 

Prüf doch auch mal, ob nach dem Einstecken des LAN-Kabels am Board an dessen Ethernet-Anschluß die beiden LEDs zu leuchten und blinken beginnen.
